scripts/utils/trading_strategy_helpers.js

```diff
@@ -151,20 +153,25 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
    * @return {Promise<TokenObject>[]} list of detailed/relevant token information
    */
   const fetchTokenInfoFromExchange = function (exchange, tokenIds, debug = false) {
-    const log = debug ? () => console.log.apply(arguments) : () => {}
+    const log = debug ? (...a) => console.log(...a) : () => {}
 
-    log("Fetching token data from EVM")
+    let requiresFetching = false
     const tokenPromises = {}
     for (const id of tokenIds) {
       if (!(id in globalTokenPromisesFromId)) {
+        requiresFetching = true
         globalTokenPromisesFromId[id] = (async () => {
           const tokenAddress = await exchange.tokenIdToAddressMap(id)
-          log(`Token id ${id} corresponds to token at address ${tokenAddress}`)
-          return fetchTokenInfoAtAddresses([tokenAddress], debug)[tokenAddress]
+          const tokenInfo = await fetchTokenInfoAtAddresses([tokenAddress], false)[tokenAddress]
+          log(
+            `Found token ${tokenInfo.symbol} for exchange id ${id} at address ${tokenInfo.address} with ${tokenInfo.decimals} decimals`
```

Comment:
I see you are really enjoying this type of string formatting!

Reply:
Now that I'm aware it exists, I'll never use anything else if I have to print a variable!